packages/python/build

```diff
@@ -18,6 +18,12 @@ rm -rf "$PKG_PATH/lib/python3.4/test"
 # mkpanda option to apply to any package.
 find "$PKG_PATH/lib/" ! -type d -name "*.a" -exec rm -f -- '{}' +
 
+# This works around the issue of Python's build not being able to hardlink these files.
+# This is due to the build being performed inside of the VM's shared folder.
+# https://www.virtualbox.org/ticket/818
+if [[ "$OS" == "darwin" ]]; then
+    cp -f "$PKG_PATH/bin/python3.4m" "$PKG_PATH/bin/python3.4"
```

Comment:
I ran into a problem with this change.

```
cp: '/opt/mesosphere/packages/python--2de80c630ae82c807c0a63d24a6eed8640faf2f6/bin/python3.4m' and '/opt/mesosphere/packages/python--2de80c630ae82c807c0a63d24a6eed8640faf2f6/bin/python3.4' are the same file
ERROR Building packages: docker exited non-zero: 1
Command: docker run -v /Users/senthil/workspace/dcos/dcos/packages/cache/packages/python/src:/pkg/src:rw -v /Users/senthil/workspace/dcos/dcos/packages/python/build:/pkg/build:ro -v /Users/senthil/workspace/dcos/dcos/packages/cache/packages/python/result:/opt/mesosphere/packages/python--2de80c630ae82c807c0a63d24a6eed8640faf2f6:rw -v /Users/senthil/pkgpanda/pkgpanda-ba9bfkxu:/opt/mesosphere:ro -v /Users/senthil/pkgpanda/pkgpanda_repor5ozy_h4/openssl--b549a917aa074e6fa349fbb15c405d074b3d3b8c:/opt/mesosphere/packages/openssl--b549a917aa074e6fa349fbb15c405d074b3d3b8c:ro -e PKG_VERSION=2de80c630ae82c807c0a63d24a6eed8640faf2f6 -e PKG_NAME=python -e PKG_PATH=/opt/mesosphere/packages/python--2de80c630ae82c807c0a63d24a6eed8640faf2f6 -e OS=darwin -e PKG_VARIANT=<default> -e PKG_ID=python--2de80c630ae82c807c0a63d24a6eed8640faf2f6 dcos/dcos-builder:dcos-builder_dockerdir-latest /bin/bash -o nounset -o pipefail -o errexit /pkg/build
Traceback (most recent call last):
  File "/Users/senthil/workspace/dcos/dcos/pkgpanda/build/__init__.py", line 1184, in build
    "/pkg/build"])
  File "/Users/senthil/workspace/dcos/dcos/pkgpanda/build/__init__.py", line 51, in run
    check_call(docker)
  File "/Users/senthil/python3.4/lib/python3.4/subprocess.py", line 558, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['docker', 'run', '-v', '/Users/senthil/workspace/dcos/dcos/packages/cache/packages/python/src:/pkg/src:rw', '-v', '/Users/senthil/workspace/dcos/dcos/packages/python/build:/pkg/build:ro', '-v', '/Users/senthil/workspace/dcos/dcos/packages/cache/packages/python/result:/opt/mesosphere/packages/python--2de80c630ae82c807c0a63d24a6eed8640faf2f6:rw', '-v', '/Users/senthil/pkgpanda/pkgpanda-ba9bfkxu:/opt/mesosphere:ro', '-v', '/Users/senthil/pkgpanda/pkgpanda_repor5ozy_h4/openssl--b549a917aa074e6fa349fbb15c405d074b3d3b8c:/opt/mesosphere/packages/openssl--b549a917aa074e6fa349fbb15c405d074b3d3b8c:ro', '-e', 'PKG_VERSION=2de80c630ae82c807c0a63d24a6eed8640faf2f6', '-e', 'PKG_NAME=python', '-e', 'PKG_PATH=/opt/mesosphere/packages/python--2de80c630ae82c807c0a63d24a6eed8640faf2f6', '-e', 'OS=darwin', '-e', 'PKG_VARIANT=<default>', '-e', 'PKG_ID=python--2de80c630ae82c807c0a63d24a6eed8640faf2f6', 'dcos/dcos-builder:dcos-builder_dockerdir-latest', '/bin/bash', '-o', 'nounset', '-o', 'pipefail', '-o', 'errexit', '/pkg/build']' returned non-zero exit status 1

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/tmp/dcos_build_venv/bin/release", line 9, in <module>
    load_entry_point('dcos-image', 'console_scripts', 'release')()
  File "/Users/senthil/workspace/dcos/dcos/release/__init__.py", line 813, in main
    release_manager.create('testing', options.channel, options.tag)
  File "/Users/senthil/workspace/dcos/dcos/release/__init__.py", line 718, in create
    metadata = make_stable_artifacts(self.__config['options']['cloudformation_s3_url'] + '/' + repository_path)
  File "/Users/senthil/workspace/dcos/dcos/release/__init__.py", line 302, in make_stable_artifacts
    all_completes = do_build_packages(cache_repository_url)
  File "/Users/senthil/workspace/dcos/dcos/release/__init__.py", line 519, in do_build_packages
    result = pkgpanda.build.build_tree(package_store, True, None)
  File "/Users/senthil/workspace/dcos/dcos/pkgpanda/build/__init__.py", line 670, in build_tree
    True)
  File "/Users/senthil/workspace/dcos/dcos/pkgpanda/build/__init__.py", line 1186, in build
    raise BuildError("docker exited non-zero: {}\nCommand: {}".format(ex.returncode, ' '.join(ex.cmd)))
pkgpanda.build.BuildError: docker exited non-zero: 1
Command: docker run -v /Users/senthil/workspace/dcos/dcos/packages/cache/packages/python/src:/pkg/src:rw -v /Users/senthil/workspace/dcos/dcos/packages/python/build:/pkg/build:ro -v /Users/senthil/workspace/dcos/dcos/packages/cache/packages/python/result:/opt/mesosphere/packages/python--2de80c630ae82c807c0a63d24a6eed8640faf2f6:rw -v /Users/senthil/pkgpanda/pkgpanda-ba9bfkxu:/opt/mesosphere:ro -v /Users/senthil/pkgpanda/pkgpanda_repor5ozy_h4/openssl--b549a917aa074e6fa349fbb15c405d074b3d3b8c:/opt/mesosphere/packages/openssl--b549a917aa074e6fa349fbb15c405d074b3d3b8c:ro -e PKG_VERSION=2de80c630ae82c807c0a63d24a6eed8640faf2f6 -e PKG_NAME=python -e PKG_PATH=/opt/mesosphere/packages/python--2de80c630ae82c807c0a63d24a6eed8640faf2f6 -e OS=darwin -e PKG_VARIANT=<default> -e PKG_ID=python--2de80c630ae82c807c0a63d24a6eed8640faf2f6 dcos/dcos-builder:dcos-builder_dockerdir-latest /bin/bash -o nounset -o pipefail -o errexit /pkg/build
/Users/senthil/python3.4/lib/python3.4/tempfile.py:699: ResourceWarning: Implicitly cleaning up <TemporaryDirectory '/Users/senthil/pkgpanda/pkgpanda_repor5ozy_h4'>
  _warnings.warn(warn_message, ResourceWarning)

```

I had to revert this portion of change to proceed.